Sadot/User/UserHomePage.cs

```diff
@@ -49,16 +88,48 @@ private void FillTableList()
             tables = db.GetTablesData();
             for (int i = 0; i < tables.Length; i++)
             {
-                dgvTableList.Rows.Add(tables[i].TableID, tables[i].TableStatus);
+                if (tables[i].OrderState == "הזמנה בהכנה")
+                {
+                    DateTime waitingTime = DateTime.Now - tables[i].TimeOfOrder.TimeOfDay;
+                    dgvTableList.Rows.Add(tables[i].TableID, tables[i].TableStatus, tables[i].OrderState, waitingTime.ToString("mm:ss"));
+                }
+                else
+                    dgvTableList.Rows.Add(tables[i].TableID, tables[i].TableStatus, tables[i].OrderState, tables[i].TimeOfOrder.ToString("mm:ss"));
+
                 dgvTableList.Rows[i].Selected = false;
+
                 if (tables[i].TableStatus == "פנוי")
-                    FiilRowColor(Color.White, i);
+                {
+                    if (tables[i].OrderState == "NA")
+                    {
+                        FiilRowColor(Color.White, i);
+                        tableButtons[tables[i].TableID].BackColor = System.Drawing.Color.White;
+                    }
+                    else
+                    {
+                        FiilRowColor(Color.Yellow, i);
+                        tableButtons[tables[i].TableID].BackColor = System.Drawing.Color.Yellow;
+                    }
+                    
+                }
                 else if (tables[i].TableStatus == "תפוס")
-                    FiilRowColor(Color.Green, i);
+                {
+                    if(tables[i].OrderState == "הזמנה בהכנה")
+                    {
+                        FiilRowColor(Color.Red, i);
+                        tableButtons[tables[i].TableID].BackColor = System.Drawing.Color.Red;
+                    }
+                    else
+                    {
+                        FiilRowColor(Color.Green, i);
+                        tableButtons[tables[i].TableID].BackColor = System.Drawing.Color.Green;
+                    }
+                }             
                 else if (tables[i].TableStatus == "בחשבון")
+                {
                     FiilRowColor(Color.SkyBlue, i);
-                else
-                    FiilRowColor(Color.Yellow, i);
```

Comment:
Instead of writing this:
**FiilRowColor(Color.White, i);
tableButtons[tables[i].TableID].BackColor = System.Drawing.Color.White;**
multiple times, you can asign only **Color color** and **Color backColor**
and at the end write:
**FiilRowColor(color, i);
tableButtons[tables[i].TableID].BackColor = backColor;**

Reply:
fixed.
I have rewritten the function - review it again, please